analysis/analyze-hotspots/build.gradle

```
@@ -36,6 +36,18 @@ dependencies {
     natives "com.esri.arcgisruntime:arcgis-java-resources:$arcgisVersion"
 }
 
+task apiKey {
+    description = "Creates a new gradle.properties file with an empty API key variable in the user home ./gradle folder if the file doesn't already exist."
+    group = "build"
+    def apiKeyFile = new File("${System.properties.getProperty("user.home")}/.gradle/gradle.properties")
+    if (!apiKeyFile.exists()) {
+        print("Go to " + new URL("https://developers.arcgis.com") + " to get an API key.")
+        print(" Add your API key to ${System.properties.getProperty("user.home")}\\.gradle\\gradle.properties.")
+        String apiKeyFileContents = "apiKey = "
+        apiKeyFile.write(apiKeyFileContents)
```

Technically it's a properties file, and it could have other uses than just the Api key. Should we rename perhaps?

```suggestion
    def propertiesFile = new File("${System.properties.getProperty("user.home")}/.gradle/gradle.properties")
    if (!propertiesFile.exists()) {
        print("Go to " + new URL("https://developers.arcgis.com") + " to get an API key.")
        print(" Add your API key to ${System.properties.getProperty("user.home")}\\.gradle\\gradle.properties.")
        String propertiesFileContents = "apiKey = "
        propertiesFile.write(propertiesFileContents)
```

The last two lines could also possibly be written in one line?
`propertiesFile.write("apiKey = ")`?
I'm not 100% it works though, so worth trying it out.

---

It does work!